// Copyright (c) 2023 Boston Dynamics, Inc.  All rights reserved.
//
// Downloading, reproducing, distributing or otherwise using the SDK Software
// is subject to the terms and conditions of the Boston Dynamics Software
// Development Kit License (20191101-BDSDK-SL).

syntax = "proto3";

package bosdyn.api;

option java_outer_classname = "LicenseServiceProto";

import "bosdyn/api/license.proto";

// The LicenseService allows clients to query the currently installed license on robot.
service LicenseService {
    // Get information, such as the license number, dates of validity, and features for the license
    // currently uploaded on the robot.
    rpc GetLicenseInfo(GetLicenseInfoRequest) returns (GetLicenseInfoResponse) {}

    // Check if specific features (identified by string names) are enabled under the currently loaded
    // license for this robot.
    rpc GetFeatureEnabled(GetFeatureEnabledRequest) returns (GetFeatureEnabledResponse) {}
}
